tests/exporter/exporter_test.go

```diff
@@ -0,0 +1,247 @@
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	csp "github.com/openebs/maya/pkg/cstorpool/v1alpha3"
+	cv "github.com/openebs/maya/pkg/cstorvolume/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	spc "github.com/openebs/maya/pkg/storagepoolclaim/v1alpha1"
+	"github.com/openebs/maya/tests"
+	framework "github.com/openebs/maya/tests/framework/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("Test maya-exporter [single-pool-pod]", func() {
+	var (
+		err error
+		pod *corev1.PodList
+	)
+	BeforeEach(func() {
+		When("we are creating pool deployment", func() {
+			By("building object of storageclass")
+			scObj, err = sc.NewBuilder().
+				WithName(scName).
+				WithAnnotations(annotations).
+				WithProvisioner(openebsProvisioner).Build()
+			Expect(err).ShouldNot(HaveOccurred(), "while building storageclass obj for storageclass {%s}", scName)
+
+			By("creating storageclass")
+			_, err = ops.SCClient.Create(scObj)
+			Expect(err).To(BeNil(), "while creating storageclass", scName)
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(1).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count")
+			Eventually(func() int {
+				cspAPIList, err = ops.CSPClient.List(metav1.ListOptions{})
+				Expect(err).To(BeNil())
+				count := csp.
+					ListBuilderForAPIObject(cspAPIList).
+					List().
+					Filter(csp.HasLabel(string(apis.StoragePoolClaimCPK), spcName), csp.IsStatus("Healthy")).Len()
+				return count
+			},
+				framework.DefaultTimeOut, framework.DefaultPollingInterval).
+				Should(Equal(1), "while getting healthy csp count")
+
+			By("listing cstor pool pods")
+			selector := map[string]string{
+				string(apis.StoragePoolClaimCPK): spcName,
+				"app": "cstor-pool",
+			}
+
+			ls := labels.Set(selector).
+				AsSelector().
+				String()
+			pod, err = ops.PodClient.List(metav1.ListOptions{
+				LabelSelector: ls,
+			})
+
+			Expect(err).To(BeNil(), "while listing pool pods with selector ", ls)
+
+			By("verifying pod items")
+			Expect(len(pod.Items)).To(Equal(1), "while getting pod items length", pod)
+
+			By("verifying no of containers in pool pod")
+			Expect(len(pod.Items[0].Spec.Containers)).To(Equal(3), "while getting no of containers", pod)
+
+			By("verifying whether maya-exporter container exists")
+			Expect(pod.Items[0].Spec.Containers[2].Name).To(Equal("maya-exporter"), "while verifying container name", pod)
```

Comment:
IMO instead of using indexing better to go with `Expect(pod.Items[0].Spec.Containers[*].Name).To(ContainSubstring("maya-exporter")) ` or
`Expect(pod.Items[0].Spec.Containers[*].Name).To(MatchRegexp("maya-exporter")) `
If we are getting `pod.Items[0].Spec.Containers[*].Name` as a string.
If we are getting `pod.Items[0].Spec.Containers[*].Name` as []string then make use of `ConsistOf` or `ContainElement`